From: Cliff Wickman <cpw@sgi.com>
Subject: x86/uv: Fix uninitialized spinlocks
Git-commit: d2ebc71d472020bc30e29afe8c4d2a85a5b41f56
Patch-mainline: Yes
References: bnc#743212

Initialize two spinlocks in tlb_uv.c.

The lack of explicit initialization seems to be functionally harmless (the
locks are initialized to 0's), but it is diagnosed when these are turned on:
        CONFIG_DEBUG_SPINLOCK=y
        CONFIG_DEBUG_MUTEXES=y
        CONFIG_DEBUG_LOCK_ALLOC=y
        CONFIG_LOCKDEP=y

Signed-off-by: Cliff Wickman <cpw@sgi.com>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/platform/uv/tlb_uv.c |    2 ++
 1 file changed, 2 insertions(+)

Index: linux-3.0-SLE11-SP2/arch/x86/platform/uv/tlb_uv.c
===================================================================
--- linux-3.0-SLE11-SP2.orig/arch/x86/platform/uv/tlb_uv.c
+++ linux-3.0-SLE11-SP2/arch/x86/platform/uv/tlb_uv.c
@@ -1851,6 +1851,8 @@ static void __init init_per_cpu_tunables
 		bcp->cong_reps			= congested_reps;
 		bcp->cong_period		= congested_period;
 		bcp->clocks_per_100_usec =	usec_2_cycles(100);
+		spin_lock_init(&bcp->queue_lock);
+		spin_lock_init(&bcp->uvhub_lock);
 	}
 }
 
